default/python/AI/empire/buildings_locations.py

```diff
@@ -29,5 +29,8 @@ def set_building_locations(
 ):
     this_grade_facilities = _get_facilities().setdefault(weapons_grade, {})
     for building in buildings_here:
-        if building in AIDependencies.SHIP_FACILITIES:
-            this_grade_facilities.setdefault(building, set()).add(pid)
+        try:
+            Shipyard(building)
+        except ValueError:
+            continue
```

Comment:
I don't like using exceptions that way, how about:
`if building in (yard.value for yard in Shipyard):`

Reply:
This approach is pretty common in Python.
https://docs.python.org/3/glossary.html#term-EAFP